RocketChat/Rocket.Chat
# [FIX] Cannot create a discussion from top left sidebar as a user

## Proposed changes (including videos or screenshots)


When trying to create a discussion using the top left sidebar modal with an role that don't have the `view-other-user-channels ` permission, an empty list would be shown, which is a wrong behavior.
Also, when being able to use this modal, discussions were listed as options, which is also a wrong behavior as there can't be nested discussions.

This PR looks to fix both these issues.

**Old behavior:**
![old](https://user-images.githubusercontent.com/49413772/124960017-3c333280-dff2-11eb-86cd-b2638311517e.png)

**New behavior:**
![image](https://user-images.githubusercontent.com/49413772/124958882-05a8e800-dff1-11eb-8203-b34a4f1c98a0.png)






## Issue(s)

[Task - ClickUp](https://app.clickup.com/t/zdfvmu)
Fixes #22534

## Steps to test or reproduce


1. Start a fresh install
2. Create an admin user and a normal user
3. As the admin, try to create a discussion from the top left sidebar, it will work (you will see the general channel in the channel list)
4. As a user, try to create a discussion from the top left sidebar, it will not work (the channel list will remains empty)

## Further comments
